### PR TITLE
Add 'Focus to terminal X' action

### LINKED ISSUE #20133: Change focus to terminal panes via number keyboard shortcut
- VSCode Version: 1.9.0
- OS Version: OSX 10.11

Steps to Reproduce:

1. Open the terminal, create a few panes
2. Try to focus a particular one with a keyboard shortcut.  I see there is `workbench.action.terminal.focusNext` but nothing to directly access a pane by number even though the dropdown contains numbers.

### PR DESCRIPTION
**Pull request by hun1ahpu**
Fixes #20133

### CODE REVIEW

File: src/vs/workbench/parts/terminal/electron-browser/terminal.contribution.ts [hunk 1]
```diff
@@ -200,9 +200,12 @@ actionRegistry.registerWorkbenchAction(new SyncActionDescriptor(CreateNewTermina
 	primary: KeyMod.CtrlCmd | KeyMod.Shift | KeyCode.US_BACKTICK,
 	mac: { primary: KeyMod.WinCtrl | KeyMod.Shift | KeyCode.US_BACKTICK }
 }), 'Terminal: Create New Integrated Terminal', category);
-actionRegistry.registerWorkbenchAction(new SyncActionDescriptor(FocusTerminalAction, FocusTerminalAction.ID, FocusTerminalAction.LABEL), 'Terminal: Focus Terminal', category);
+actionRegistry.registerWorkbenchAction(new SyncActionDescriptor(FocusActiveTerminalAction, FocusActiveTerminalAction.ID, FocusActiveTerminalAction.LABEL), 'Terminal: Focus Active Terminal', category);
 actionRegistry.registerWorkbenchAction(new SyncActionDescriptor(FocusNextTerminalAction, FocusNextTerminalAction.ID, FocusNextTerminalAction.LABEL), 'Terminal: Focus Next Terminal', category);
 actionRegistry.registerWorkbenchAction(new SyncActionDescriptor(FocusPreviousTerminalAction, FocusPreviousTerminalAction.ID, FocusPreviousTerminalAction.LABEL), 'Terminal: Focus Previous Terminal', category);
+for (let i = 0; i < 10; i++) {
```
Comment (Tyriar) [inline src/vs/workbench/parts/terminal/electron-browser/terminal.contribution.ts]: This should be 1 to 9, not 0 to 9

**Review by Tyriar — CHANGES_REQUESTED** [review]
Looks great, just a minor comment :+1:

File: src/vs/workbench/parts/terminal/electron-browser/terminal.contribution.ts (hunk 1 above)
Comment (hun1ahpu) [inline src/vs/workbench/parts/terminal/electron-browser/terminal.contribution.ts]: yep thanks

File: src/vs/workbench/parts/terminal/electron-browser/terminal.contribution.ts [hunk 2]
```diff
@@ -147,7 +147,7 @@ configurationRegistry.registerConfiguration({
 				CreateNewTerminalAction.ID,
 				CopyTerminalSelectionAction.ID,
 				KillTerminalAction.ID,
-				FocusTerminalAction.ID,
+				FocusActiveTerminalAction.ID,
```
Comment (hun1ahpu) [inline src/vs/workbench/parts/terminal/electron-browser/terminal.contribution.ts]: I wonder if all the new commands should be added here as well?

Comment (Tyriar) [inline src/vs/workbench/parts/terminal/electron-browser/terminal.contribution.ts]: Yes they should, if not the terminal may eat the keystrokes before they reach the vscode keybinding system.

**Comment by Tyriar** [comment]
@hun1ahpu tests are failing on Travis:

```
out-build/vs/workbench/parts/terminal/electron-browser/terminalActions.nls.js:13
	`Focus Terminal ${n}`,
	                  ^
ReferenceError: n is not defined
    at Object.<anonymous> (out-build/vs/workbench/parts/terminal/electron-browser/terminalActions.nls.js:13:20)
    at NodeScriptLoader._loadAndEvalScript (evalmachine.<anonymous>:696:15)
    at evalmachine.<anonymous>:643:31
    at tryToString (fs.js:455:3)
    at FSReqWrap.readFileAfterClose [as oncomplete] (fs.js:442:12)
The command "gulp optimize-vscode --silent --max_old_space_size=4096" exited with 1.
```

https://travis-ci.org/Microsoft/vscode/jobs/203133143#L3225

Maybe make it a regular static function instead of an variable storing an inline function? Also changing it to `getLabel`/`getId` is probably a good idea.

File: src/vs/workbench/parts/terminal/electron-browser/terminalActions.ts [hunk 3]
```diff
@@ -141,6 +141,34 @@ export class FocusNextTerminalAction extends Action {
...
+		this.terminalService.setActiveInstanceByIndex(this.getTerminalNumber() - 1);
+		return this.terminalService.showPanel(true);
+	}
+
+	public static getId(n: number): string {
+		return FocusTerminalByNumberAction.ID_PREFIX + n;
+	}
+
+	public static getLabel(n: number): string {
+		return nls.localize(`workbench.action.terminal.focusByNumber${n}`, `Focus Terminal ${n}`, n);
```
Comment (Tyriar) [inline src/vs/workbench/parts/terminal/electron-browser/terminalActions.ts]: Almost there 😃 I think this should be:

```ts
return nls.localize('workbench.action.terminal.focusByNumber', 'Focus Terminal {0}', n);
```

That way our localization system will understand it.

**Review by Tyriar — APPROVED** [review]

**Comment by Tyriar** [comment]
Thanks @hun1ahpu!

**Pull request merged**